### ISSUE TITLE
support for pdf plot output

### ISSUE THREAD

**Issue by jjallaire** [JuliaLang/IJulia.jl#1035]
Hi there, we've been putting some work into integrating support for IJulia into Quarto (<https://quarto.org>), which is a scientific publishing system based on Pandoc (sort of like R Markdown, in fact built by the same team, but fundamentally cross-language and cross-engine).

We have plots working correctly w/ various output formats, but one noteworthy limitation is that when creating a PDF the plot output is provided as PNG (as SVG currently doesn't work out of the box w/ LaTeX). As you can imagine this results in dramatically lower quality than is possible with native PDF plot output. One of Quarto's main uses cases is creation of scientific publications for academic journals, so getting high quality output ends up being pretty important.

I see the available output formats in IJulia enumerated here:

https://github.com/JuliaLang/IJulia.jl/blob/cc2a9bf61a2515596b177339f9a3514de8c38573/src/display.jl#L19-L28

Further, I noted the place where JuliaPlots is yielding output here:

https://github.com/JuliaPlots/Plots.jl/blob/master/src/ijulia.jl#L33-L56

It would be fantastic if these could be updated to produce PDF output. This might not at first blush seem to make sense for a  "notebook" but if you think of Jupyter as an execution engine for realizing executable code into output it makes more sense (i.e. think about `nbconvert` as the client rather than the traditional browser front end). This does in fact currently work for Matplotlib output on the Python side.

It's certainly a fine stopgap to use PNG output for plots in PDFs, but my guess is that when users start creating articles they want to submit for publication that the PDF support will quickly go from nice-to-have to must have.

**Comment by jjallaire** [comment]
Once this lands we still might need to change Quarto to stop "redirecting" pdf to png here: https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/jupyter/lang/julia/setup.jl#L22-L27

Further, we might want some sort of version check to make sure that we use the workaround when updated IJulia isn't avaialble.

**Comment by jeremiahpslewis** [comment]
Ooh, good that you mention a workaround for earlier versions of IJulia (once/if PR is merged). I wanted to suggest a slightly different approach: what about a Quarto.jl package which bundles the logic which is currently in the setup.jl file, requires a couple of packages (Plots, IJulia), and sets a lower bound on the packages based on PDF support (and other things as they may come up in the future).

**Comment by jjallaire** [comment]
I think we need to keep the code in `setup.jl` in sync with the current version of Quarto (there can be dependencies both ways) so it's much better to have this be embedded (otherwise the package will need to have quarto version conditional code and quarto will need to have package version conditional code)

**Comment by jeremiahpslewis** [comment]
I think this can be closed, see discussion here: https://github.com/JuliaLang/IJulia.jl/pull/1038

**Comment by jeremiahpslewis** [comment]
This PR to quarto will (when ready) resolve this issue https://github.com/quarto-dev/quarto-cli/pull/800
